src/agent/src/rpc.rs

```diff
@@ -1407,7 +1407,14 @@ fn get_memory_info(block_size: bool, hotplug: bool) -> Result<(u64, bool)> {
                     return Err(anyhow!("Invalid block size"));
                 }
 
-                size = v.trim().parse::<u64>()?;
+                size = match u64::from_str_radix(v.trim(), 16) {
+                    Ok(h) => h,
+                    Err(_) => {
+                        warn!(sl!(), "failed to parse the str {} to hex\n", size);
```

Comment:
hi, @egernst the `\n` has some special meanings?

Reply:
Just for adding a new line? I referenced/followed how it was done @ https://github.com/kata-containers/kata-containers/pull/910/files/9777a587450bb86401b5e7d0320c75688acc836f#diff-ed6985bdc917c7f40cb7b17412af3a5dL1642

Reply:
There is an implicit new line at the end of each warn!() call. See the other `info!` and `warn!` in the same file.

Reply:
Will remove.  Seems its a bit inconsistent in the file.